FiveCalls/FiveCalls/IssuesViewController.swift

```diff
@@ -118,13 +128,26 @@ class IssuesViewController : UITableViewController {
         lastLoadResult = result
         issuesDelegate?.didFinishLoadingIssues()
         if case .success = result {
-            viewModel = ViewModel(issues: issuesManager.issues)
+            viewModel = ViewModel(categories: issuesManager.categories)
```

Comment:
Here we do have _all_ the issues, but we're handing over a specific view of those issues by grouping them into categories. What if we moved this categorization/group _into_ the view model, then we'd still have access to the entire array of issues for the Active Issues mode, and the grouped for the All Issues list.

Reply:
I don't think I understand you.

In this PR - the table view always has as many section as the number of unique categories. Its just that when we want to show active issues - section headers are not shown. So if there are 7 categories in the response - the number of sections displayed by the tableView would be 7. But in the viewForHeaderInSection we return 'nil' because we don't want to group issues by category. As a result Issues appear as if they are in a single section.

So currently we are handing over the entire <i>view</i> to the view model - not a subset of it. Also each Category object (inside the categories array) contains all the issues that belong to it - active and inactive both. If a specific Issue inside a category is inactive and if we are currently not showing all issues - the cell is hidden. 

Also if we pass the response (all issues) as is to the ViewModel and let it process it (sort, associate categories), that work would be done on the main thread - which is not ideal (again I am not sure if I understand your original comment, so this last paragraph could be illogical)